### PR TITLE
Remove dependency on AWS SDK v2 and use modularized AWS SDK v3

### PR DESCRIPTION
**Pull request by PlasmaIntec**
*Issue #, if available:*
#93 

*Description of changes:*
Switch to using AWS SDK v3.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by amzn-paunort** [comment]
This PR should probably also bump our local major version, or at least a minor version if the exposed API has not changed.

**Comment by amzn-paunort** [comment]
> This PR should probably also bump our local major version, or at least a minor version if the exposed API has not changed.

Looks like this has been done now!

File: src/QldbDriver.ts [hunk 1]
```diff
@@ -178,57 +182,60 @@ export class QldbDriver {
...
+                return false;
             }
-            return false;
         }
         
         retryConfig = (retryConfig == null) ? this._retryConfig : retryConfig;
-        let session: QldbSession;
-        let startNewSession: boolean = false;
+        let replaceDeadSession: boolean = false;
+        let retryAttempt = 0;
```
Comment (allimn) [inline src/QldbDriver.ts:230]: L230 should be removed.

File: src/Communicator.ts [hunk 2]
```diff
@@ -84,14 +83,14 @@ export class Communicator {
      * @param commitDigest The digest hash of the transaction to commit.
      * @returns Promise which fulfills with the commit transaction response returned from QLDB.
      */
-    async commit(txnId: string, commitDigest: CommitDigest): Promise<CommitTransactionResult> {
-        const request: SendCommandRequest = {
+    async commit(txnId: string, commitDigest: Uint8Array | undefined): Promise<CommitTransactionResult> {
```
Comment (allimn) [inline src/Communicator.ts:86]: CommitDigest?

Comment (PlasmaIntec) [reply src/Communicator.ts:86]: CommitDigest is not available in [@aws-sdk/client-qldb-session](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-qldb-session/index.html)

Comment (allimn) [reply src/Communicator.ts:86]: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-qldb-session/modules/committransactionrequest.html#commitdigest

**Pull request merged**